test/HalResourceTest.php

```diff
@@ -235,13 +239,22 @@ public function testWithElementProxiesToEmbedIfResourceCollectionValueProvided()
         $this->assertEquals(['foo' => $collection], $new->getElements());
     }
 
-    public function testWithElementDoesNotProxyToEmbedIfAnEmptyArrayValueIsProvided(): void
+    public function testWithElementWillEmbedAnEmptyArrayIfAnEmptyArrayValueIsProvided(): void
```

Comment:
```suggestion
    public function testWithElementWillEmbedAnEmptyArrayIfAnEmptyArrayValueIsProvidedAndConfiguredToEmbedEmptyCollections(): void
```

Since the default behavior remains unchanged, we should note in the test that the tested behavior is specifically for when the embed empty collections option is toggled on.

Reply:
Also... it looks like you removed the tests that test the default behavior; can you please reinstate those?